drivers/dma/Kconfig.mcux_lpc

```diff
@@ -18,4 +18,17 @@ config DMA_NUMBER_OF_DESCRIPTORS
 	  Increase or decrease this value depending on the max number of data
 	  transferred by the application.
 
+config DMA_NUMBER_OF_CHANNELS_ALLOCATED
```

Comment:
I'm not sure why we didn't do this for the other Kconfig in the driver, but most driver specific Kconfigs have some sort of vendor name included, IE `DMA_LPC_NUMBER_OF_CHANNELS_ALLOCATED`. Just a suggestion, feel free to leave the name here unchanged.

Reply:
yes I noticed this but was just trying to be consistent

Reply:
I updated the PR renaming all the Kconfigs